controllers/vspheredeploymentzone_controller.go

```diff
@@ -160,7 +149,13 @@ func (r vsphereDeploymentZoneReconciler) Reconcile(ctx context.Context, request
 }
 
 func (r vsphereDeploymentZoneReconciler) reconcileNormal(ctx context.Context, deploymentZoneCtx *capvcontext.VSphereDeploymentZoneContext) error {
-	authSession, err := r.getVCenterSession(ctx, deploymentZoneCtx)
+	failureDomain := &infrav1.VSphereFailureDomain{}
+	failureDomainKey := client.ObjectKey{Name: deploymentZoneCtx.VSphereDeploymentZone.Spec.FailureDomain}
+	// Return an error if the FailureDomain can not be retrieved.
+	if err := r.Client.Get(ctx, failureDomainKey, failureDomain); err != nil {
```

Comment:
could also set status on the vspheredeploymentzone here, but I didn't want to increase the scope of this PR again.